File: multi/domain/src/main/java/com/example/domain/movies/entity/Movie.java
package com.example.domain.movies.entity;

import com.example.domain.common.BaseEntity;
import com.example.domain.movies.entity.enums.Genre;
import com.example.domain.movies.entity.enums.Rating;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "Movies")
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
public class Movie extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "movie_id")
    private Long id;

    @Column(name = "title")
    private String title;

    @Column(name = "rating")
    @Enumerated(EnumType.STRING)
    private Rating rating;

    @Column(name = "release_date")
    private LocalDateTime releaseDate;

    @Column(name = "thumbnail_url")
    private String thumbnailUrl;

    @Column(name = "running_time")
    private Integer runningTime;

    @Column(name = "genre")
    @Enumerated(EnumType.STRING)
    private Genre genre;

}

File: src/main/kotlin/com/example/demo/domain/entity/Movie.kt
package com.example.demo.domain.entity

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "movies")
data class Movie(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @Column(nullable = false, length = 255)
    val title: String,

    @Enumerated(EnumType.STRING)
    val rating: Rating,

    val releaseDate: LocalDateTime,

    val thumbnailUrl: String,

    val runningTime: Int,

    val genre: String
)

enum class Rating {
    GENERAL, PG13, R, NC17
} 